juriscraper/opinions/united_states/state/nh.py

```diff
@@ -1,127 +1,58 @@
-                continue
-            dates.extend([case_date] * len(element.xpath(sub_path)))
-        return dates
-
-    def _get_precedential_statuses(self):
-        return ["Published"] * len(self.case_names)
-
-    def _get_docket_numbers(self):
-        return [pair["docket"] for pair in self.docket_name_pairs]
-
-    def _get_anchor_docket_name_pairs(self):
-        """The court has some ugly HTML practices that we need to handle.
-
-        Most anchor links include single line strings with a single docket
-        number and a single case name.  However, there are two other formats
-        we've seen and must work around.
-
-        (CASE 1)
-        The anchor has multiple lines broken with <br> tag(s), and each
-        line contains "<docket> <name>". In this case we need to combine
-        the docket numbers and name strings respectively.
-        [EXAMPLE: February 18, 2016 in nh_example_2.html]
-
-        (CASE 2)
-        The anchor has multiple lines broken with <br> tag(s), and the
-        second line is a continuation of a long case name started on the first
-        line.  So, the second line does not lead with a docket number, thus
-        this line's string should be glued onto the <name> substring extracted
-        from the previous line.
-        [EXAMPLE: September 18, 2018 in nh_example_6.html]
-        """
-        pairs = []
-        for anchor in self.html.xpath(self.link_path):
-            i = 0
-            dockets = []
-            name_substrings = []
-            text_anchor = anchor.text_content()
-            text_clean = text_anchor.replace("\n", "")
-
-            for text in text_clean.split(";"):
-                text = clean_string(text)
-                match = self.link_text_regex.search(text)
-                try:
-                    docket = match.group(1)
-                    dockets.append(docket)
-                    name = match.group(2)
-                    name = " ".join(name.split())
-                    name_substrings.append(name)
-                    i += 1
-                except AttributeError:
-                    if i == 0:
-                        # docket and name (root) should be contained in first substring
-                        error = f"Invalid anchor root string format: {text}"
-                        raise InsanityException(error)
-                    # no docket in the substring, its a trailing name substring
-                    # that they broke over multiple lines, so glue it to the
-                    # previous name substring
-                    name_substrings[i - 1] += f" {text}"
-            pairs.append(
+        self.url = "https://www.courts.nh.gov/content/api/documents"
+        self.method = "GET"
+        self.parameters = {
+            "sort": "field_date_posted|desc",
+            "page": "1",
+            "size": "25",
+            "purpose": "1331+undefined",
+            "tag": "1206+",
```

Comment:
Also - a comment explaining these values may be helpful.  

I think tag is important but im not sure purpose is.

Reply:
`"purpose"` must be included because it filter out other types of documents that aren't opinions. Opinions have a `"purpose"` value of **1331|**, while other documents have values: **1856|**, **1491|** and **1691|**
For example:
`
{'documentContent': "<div class='document__detail'>\n\t\t\t\t<div class='document__detail__icon__container'>\n\t\t\t\t\t<img src='https://www.courts.nh.gov/sites/g/files/ehbemt471/files/2019-12/icon-pdf.png' aria-label='PDF Document Icon'/>\n\t\t\t\t</div>\n\t\t\t\t<div class='document__detail__container'>\n\t\t\t\t\t<div class='document__detail__title'><h3><a href='https://www.courts.nh.gov/sites/g/files/ehbemt471/files/documents/2021-11/20200490.pdf' >2020-0490, Mary Feeney v. Karyn Kelley</a></h3></div>\n\t\t\t\t\t<div class='document__detail__description'></div>\n\t\t\t\t\t<div class='document__detail__information'><div><strong>Document Format: </strong>PDF</div></div>\n\t\t\t\t\t<div class='document__detail__information'><div><strong>Date: </strong>10/20/2021</div></div>\n\t\t\t\t\t<div class='document__detail__information'></div>\n\t\t\t\t\t<div class='document__detail__information'></div>\n\t\t\t\t</div>\n\t\t\t</div>", 'documentNumber': None, 'documentJudge': None, 'documentFiled': None, 'documentPosted': '2021-10-20', 'documentRevised': None, 'documentParties': None, 'documentCategories': None, 'documentPurposes': **'1856|'**, 'documentTags': None, 'documentName': '2020-0490, Mary Feeney v. Karyn Kelley'}`

`
{'documentContent': "<div class='document__detail'>\n\t\t\t\t<div class='document__detail__icon__container'>\n\t\t\t\t\t<img src='https://www.courts.nh.gov/sites/g/files/ehbemt471/files/2019-12/icon-pdf.png' aria-label='PDF Document Icon'/>\n\t\t\t\t</div>\n\t\t\t\t<div class='document__detail__container'>\n\t\t\t\t\t<div class='document__detail__title'><h3><a href='https://www.courts.nh.gov/sites/g/files/ehbemt471/files/documents/2021-10/12-10-21-public-hearing-notice-and-order.pdf' >Public hearing notice and order seeking comment December 10, 2021 public hearing</a></h3></div>\n\t\t\t\t\t<div class='document__detail__description'></div>\n\t\t\t\t\t<div class='document__detail__information'><div><strong>Document Format: </strong>PDF</div></div>\n\t\t\t\t\t<div class='document__detail__information'><div><strong>Date: </strong>10/04/2021</div></div>\n\t\t\t\t\t<div class='document__detail__information'></div>\n\t\t\t\t\t<div class='document__detail__information'></div>\n\t\t\t\t</div>\n\t\t\t</div>", 'documentNumber': None, 'documentJudge': None, 'documentFiled': None, 'documentPosted': '2021-10-04', 'documentRevised': None, 'documentParties': None, 'documentCategories': None, 'documentPurposes': **'1491|'**, 'documentTags': None, 'documentName': 'Public hearing notice and order seeking comment December 10, 2021 public hearing'}`

`
{'documentContent': "<div class='document__detail'>\n\t\t\t\t<div class='document__detail__icon__container'>\n\t\t\t\t\t<img src='https://www.courts.nh.gov/sites/g/files/ehbemt471/files/2019-12/icon-pdf.png' aria-label='PDF Document Icon'/>\n\t\t\t\t</div>\n\t\t\t\t<div class='document__detail__container'>\n\t\t\t\t\t<div class='document__detail__title'><h3><a href='https://www.courts.nh.gov/sites/g/files/ehbemt471/files/documents/2021-09/superior-court-administrative-order-nubmer-2021-006.pdf' >Administrative Order 2021 - 06: Obsolete Administrative Orders</a></h3></div>\n\t\t\t\t\t<div class='document__detail__description'></div>\n\t\t\t\t\t<div class='document__detail__information'><div><strong>Document Format: </strong>PDF</div></div>\n\t\t\t\t\t<div class='document__detail__information'><div><strong>Date: </strong>09/30/2021</div></div>\n\t\t\t\t\t<div class='document__detail__information'></div>\n\t\t\t\t\t<div class='document__detail__information'></div>\n\t\t\t\t</div>\n\t\t\t</div>", 'documentNumber': None, 'documentJudge': None, 'documentFiled': None, 'documentPosted': '2021-09-30', 'documentRevised': None, 'documentParties': None, 'documentCategories': None, 'documentPurposes': **'1691|'**, 'documentTags': None, 'documentName': 'Administrative Order 2021 - 06: Obsolete Administrative Orders'}
`

I'll add a comment about this in the parameters for the call to the API.

Reply:
So far, I have't found an opinion file withou a `"tag"` value of **1206|**, but this parameter can also have the value **None**. I recommend to keep it.

`{'documentContent': "<div class='document__detail'>\n\t\t\t\t<div class='document__detail__icon__container'>\n\t\t\t\t\t<img src='https://www.courts.nh.gov/sites/g/files/ehbemt471/files/2019-12/icon-pdf.png' aria-label='PDF Document Icon'/>\n\t\t\t\t</div>\n\t\t\t\t<div class='document__detail__container'>\n\t\t\t\t\t<div class='document__detail__title'><h3><a href='https://www.courts.nh.gov/sites/g/files/ehbemt471/files/documents/2021-11/20210008.pdf' >2021-0008, In the Matter of Jennifer Juri and  Michael Juri</a></h3></div>\n\t\t\t\t\t<div class='document__detail__description'></div>\n\t\t\t\t\t<div class='document__detail__information'><div><strong>Document Format: </strong>PDF</div></div>\n\t\t\t\t\t<div class='document__detail__information'><div><strong>Date: </strong>10/12/2021</div></div>\n\t\t\t\t\t<div class='document__detail__information'></div>\n\t\t\t\t\t<div class='document__detail__information'></div>\n\t\t\t\t</div>\n\t\t\t</div>", 'documentNumber': None, 'documentJudge': None, 'documentFiled': None, 'documentPosted': '2021-10-12', 'documentRevised': None, 'documentParties': None, 'documentCategories': None, 'documentPurposes': '1856|', 'documentTags': None, 'documentName': '2021-0008, In the Matter of Jennifer Juri and  Michael Juri'}
`

I'll also add a comment about param value.